packages/backend/src/workers/provider/OpenVINO.ts

```diff
@@ -0,0 +1,296 @@
+        // must be the port INSIDE the container not the exposed one
+        Test: ['CMD-SHELL', `curl -sSf localhost:8000/metrics > /dev/null`],
+        Interval: SECOND * 5,
+        Retries: 4 * 5,
+      },
+      Labels: labels,
+      Env: envs,
+      Cmd: cmd,
+    };
+  }
+
+  override async prePerform(config: InferenceServerConfig): Promise<void> {
+    const modelInfo = this.validateAndGetModelInfo(config);
+
+    if (modelInfo.file === undefined) {
+      throw new Error('The model info file provided is undefined');
+    }
+
+    await this.ensureGraphFile(modelInfo.file.path);
+
+    await this.ensureConfigFile(modelInfo);
+  }
+
+  async perform(config: InferenceServerConfig): Promise<InferenceServer> {
+    const modelInfo = this.validateAndGetModelInfo(config);
+
+    if (modelInfo.file === undefined) {
+      throw new Error('The model info file provided is undefined');
+    }
+
+    let connection: ContainerProviderConnection | undefined = undefined;
+    if (config.connection) {
+      connection = this.podmanConnection.getContainerProviderConnection(config.connection);
+    } else {
+      connection = this.podmanConnection.findRunningContainerProviderConnection();
+    }
+
+    if (!connection) throw new Error('no running connection could be found');
+
+    const vmType: VMType = (connection.vmType ?? VMType.UNKNOWN) as VMType;
+
+    // pull the image
+    const imageInfo: ImageInfo = await this.pullImage(
+      connection,
+      config.image ?? this.getOpenVINOInferenceImage(vmType),
+      config.labels,
+    );
+
+    // Get the container creation options
+    const containerCreateOptions: ContainerCreateOptions = await this.getContainerCreateOptions(
+      config,
+      imageInfo,
+      modelInfo,
+    );
+
+    // Create the container
+    const { engineId, id } = await this.createContainer(imageInfo.engineId, containerCreateOptions, config.labels);
+
+    return {
+      container: {
+        engineId: engineId,
+        containerId: id,
+      },
+      connection: {
+        port: config.port,
+      },
+      status: 'running',
+      models: config.modelsInfo.map(model => this.modelsManager.getModelInfo(model.id)),
+      type: InferenceType.OPENVINO,
+      labels: containerCreateOptions.Labels ?? {},
+    };
+  }
+
+  private validateAndGetModelInfo(config: InferenceServerConfig): ModelInfo {
+    if (!this.enabled()) throw new Error('not enabled');
+
+    if (config.modelsInfo.length === 0) throw new Error('Need at least one model info to start an inference server.');
+
+    if (config.modelsInfo.length > 1) {
+      throw new Error('Currently the inference server does not support multiple models serving.');
+    }
+
+    return config.modelsInfo[0];
+  }
+
+  private async ensureGraphFile(modelFolder: string): Promise<string> {
+    // check if the file exists
+    const graphFile = `${modelFolder}/graph.pbtxt`;
+    // check if the graph file exists
+    if (!existsSync(graphFile)) {
```

Comment:
don't use existsSync if we are in a async function? let's use `import { access } from 'node:fs/promises`

Reply:
I thought we had a conversation about it and it was not clear which one is the best